### PR TITLE
E2E Testing Base Suite

### LINKED ISSUE #366: Subjects never load for imported roads
image

Error briefly shows up in dialog when importing, too quick to see but could just be duplicate name error as it is adding the road.  No errors in console and road seems to disappear on refresh.

### PR DESCRIPTION
**Pull request by georgiashay**
Provides a set of basic end-to-end tests to run PRs against

Several small changes in components to get tests working; includes 
- Putting window.location.href = '...' into a stubbable function
- Adding data-cy attributes to components so the tests can find them
- Checking for existence of tabs cookie before modifying it on page close
- Checking for cookies allowed before checking version number cookie

Also fixes #366 and #344

### CODE REVIEW

**Comment by georgiashay** [comment]
(Sorry for the excessive force-pushes, I was trying to track down an error that only happened in CI)

The new e2e-record mode records the videos to a Cypress dashboard.  It looks like only 3 accounts can have access at once for the free version.

This should be ready for review now.  There's a lot of file changes, but most of it is adding data-cy attributes so the tests can find the elements, or test fixtures in the assets folder.

File: tests/e2e/specs/audit.js [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+      .click();
+
+    // Pull up course selection menu again
+    cy.getByDataCy('auditMajorChips')
+      .type('{esc}');
+
+    // Check that Course 21M-1 is not in audit now
+    cy.getByDataCy('auditItemmajor21M-1')
+      .should('not.exist');
+
```
Comment (miriam-rittenberg) [inline tests/e2e/specs/audit.js:105]: Can you add a test for removing a major using the info menu -> remove requirement?

File: tests/e2e/specs/classinfo.js [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+
+        // Check average enrollment
+        cy.getByDataCy('cardEnrollment')
+          .should('contain', '38');
+
+        // Check average rating
+        cy.getByDataCy('cardRating')
+          .should('contain', '5.05');
+
+        // Check expecetd hours
```
Comment (miriam-rittenberg) [inline tests/e2e/specs/classinfo.js:58]: typo - expected

File: src/stores/courseData.js [hunk 3]
```diff
@@ -56,6 +60,11 @@ const store = new Vuex.Store({
     }
   },
   mutations: {
+    resetState (state) {
+      Object.assign(state, getDefaultState());
+      state.itemAdding = undefined;
```
[the remark this answers is not in the document]
Comment (georgiashay) [reply src/stores/courseData.js]: I believe you're right.  Removed those lines.  The function isn't used right now but I left it in if it needs to be used in future tests.

File: tests/e2e/plugins/index.js [hunk 4]
```diff
@@ -9,18 +9,74 @@
...
+  // Download code based on https://github.com/cypress-io/cypress-example-recipes/blob/master/examples/testing-dom__download/cypress/plugins/index.js
+  on('task', {
+    clearDownloads () {
+      console.log('Clearing folder %s', downloadDirectory);
+      // fs.rmdirSync(downloadDirectory, { recursive: true });
+      return null;
+    },
+    readRoadFile (filename) {
+      console.log('Reading road file %s', filename);
+      return '';
```
[the remark this answers is not in the document]
Comment (georgiashay) [reply tests/e2e/plugins/index.js]: Removed this function, wasn't being used (read the file within the test itself instead).  I missed it when reviewing the PR.

File: tests/e2e/specs/audit.js (hunk 1 above)
Comment (georgiashay) [reply tests/e2e/specs/audit.js:105]: Added this to the add + remove majors test

**Review by miriam-rittenberg — APPROVED** [review]
Looks good!

**Pull request merged**